tests/testsuite/build_dir.rs

```diff
@@ -609,6 +609,87 @@ fn template_workspace_manfiest_path_hash() {
     assert_exists(&p.root().join(&format!("target-dir/debug/foo{EXE_SUFFIX}")));
 }
 
+/// Verify that the {workspace-path-hash} does not changes if cargo is run from inside of
+/// a symlinked directory.
+/// The test approach is to build a project twice from the non-symlinked directory and a symlinked
+/// directory and then compare the build-dir paths.
+#[cargo_test]
+fn template_workspace_path_hash_should_handle_symlink() {
+    #[cfg(unix)]
+    use std::os::unix::fs::symlink;
+    #[cfg(windows)]
+    use std::os::windows::fs::symlink_dir as symlink;
+
+    let p = project()
+        .file("src/main.rs", r#"fn main() { println!("Hello, World!") }"#)
```

Comment:
For faster test execution time.

```suggestion
        .file("src/lib.rs", "")
```